### ISSUE TITLE
OrmLite: Bug with null constant in Where

### ISSUE THREAD

**Issue by OlegNadymov** [ServiceStack/Issues#444]
After update OrmLite to v.4.0.60 crashed all filters like that if this Where inside Sql.In function: 
```C#
... .Where(x=>NullProperty != null) ... 
```

For ```null``` constant creates SqlParameter with value = 
```C#
value(System.Collections.Generic.List`1[System.Object]).Item[1]
```

After testing I can get more details.

Simple example (I tested on SqlServer2012OrmLiteDialectProvider):
 ```C#
public class AnyObjectClass
{
    [Alias("_id")]
    public Guid? Id { get; set; }

    [Alias("_name")]
    [StringLength(250)]
    public string Name { get; set; }
}

var results = db.Select(db.From<AnyObjectClass>().Where(xx => Sql.In(xx.Identity, db.From<AnyObjectClass>().Where(y => y.Identity != null).Select(y => y.Identity.Value))));

It's generates below SQL:
exec sp_executesql N'SELECT "_name" AS "Name", "_id" AS "Identity" 
FROM "CIS"."AnyObjectClass"
WHERE "CIS"."AnyObjectClass"."_id" IN (SELECT "CIS"."AnyObjectClass"."_id" 
FROM "CIS"."AnyObjectClass"
WHERE ("CIS"."AnyObjectClass"."_id" <> @0) )'
,N'@0 nvarchar(max) ',@0=N'value(System.Collections.Generic.List`1[System.Object]).Item[1]'
```
This SQL has wrong parameter: type and value.

P.S. I can't replace y.Identity != null to y.Identity.HasValue

**Comment by mythz** [comment]
I'm tracking this down, but the issue is when the sub expression is nested inside of another lambda expression it's no longer a `null` constant, instead the expression returns it as:

```
(System.Nullable`1[System.Guid])(.Constant<System.Runtime.CompilerServices.StrongBox`1[System.Collections.Generic.List`1[System.Object]]>(System.Runtime.CompilerServices.StrongBox`1[System.Collections.Generic.List`1[System.Object]]).Value).Item[1]
```

A workaround is to extract the sub query out of the nested expression so it reports the expression is reports the `null` as constant expression, e.g:

```csharp
var inQ = db.From<AnyObjectClass>().Where(y => y.Identity != null).Select(y => y.Identity.Value);
var results = db.Select(db.From<AnyObjectClass>().Where(xx => Sql.In(xx.Identity, inQ)));
```

I'll keep working on it to handle this use-case.

**Comment by mythz** [comment]
ok this should now be fixed with this commit: https://github.com/ServiceStack/ServiceStack.OrmLite/commit/fd7c1d4820a1f794e8c196a8ba5979ad5081bb0d

This change is available from v4.0.61 that's now [available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet).

**Comment by OlegNadymov** [comment]
Yes! This is working now! 
Thanks for quick fix!
